SimCalorimetry/HGCalSimAlgos/interface/HGCalSiNoiseMap.h

```diff
@@ -66,13 +65,13 @@ class HGCalSiNoiseMap : public HGCalRadiationMap {
      @short returns the charge collection efficiency and noise
      if gain range is set to auto, it will find the most appropriate gain to put the mip peak close to 10 ADC counts
   */
-  const SiCellOpCharacteristicsCore getSiCellOpCharacteristicsCore(const HGCSiliconDetId &did,
+  const SiCellOpCharacteristicsCore getSiCellOpCharacteristicsCore(const unsigned cell_raw_Id,
                                                                    GainRange_t gain,
```

Comment:
I do not particularly like this approach. In my opinion, this goes against the **strongly typed** nature of `C++`  and reminds me of the `void *`, used and abused in the old `C`-days. 
This is also introducing unnecessary branching conditions in the code, that will end up wasting CPU cycles. They are unnecessary since the branch is fully known and determined by the correct **type** of `DetId`.
I'd therefore suggest a factorization of the part of the code that could stay untouched and in common between the 2 detectors, and use a template+specialization  to handle the cases driven by the type.

Reply:
thanks for the feedback, I can template this class 
give me a couple of days